src/Utils/Strict.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+/**
+ * This file is part of the Nette Framework (https://nette.org)
+ * Copyright (c) 2004 David Grudl (https://davidgrudl.com)
+ */
+
+namespace Nette;
+
+use Nette\Utils\ObjectMixin;
+
+
+/**
+ * Strict class (access to undeclared members throws exceptions) with 'did you mean' hints for better experience.
+ */
+trait Strict
+{
+
+	/**
+	 * Call to undefined method.
+	 * @throws MemberAccessException
+	 */
+	public function __call($name, $args)
+	{
+		$class = method_exists($this, $name) ? 'parent' : get_class($this);
+		$items = (new \ReflectionClass($this))->getMethods(\ReflectionMethod::IS_PUBLIC);
+		$hint = ($t = ObjectMixin::getSuggestion($items, $name)) ? ", did you mean $t()?" : '.';
+		throw new MemberAccessException("Call to undefined method $class::$name()$hint");
+	}
+
+
+	/**
+	 * Call to undefined static method.
+	 * @throws MemberAccessException
+	 */
+	public static function __callStatic($name, $args)
+	{
+		$rc = new \ReflectionClass(get_called_class());
+		$items = array_intersect($rc->getMethods(\ReflectionMethod::IS_PUBLIC), $rc->getMethods(\ReflectionMethod::IS_STATIC));
+		$hint = ($t = ObjectMixin::getSuggestion($items, $name)) ? ", did you mean $t()?" : '.';
+		throw new MemberAccessException("Call to undefined static method {$rc->getName()}::$name()$hint");
+	}
+
+
+	/**
+	 * Access to undeclared property.
+	 * @throws MemberAccessException
+	 */
+	public function &__get($name)
+	{
+		$rc = new \ReflectionClass($this);
+		$items = array_diff($rc->getProperties(\ReflectionProperty::IS_PUBLIC), $rc->getProperties(\ReflectionProperty::IS_STATIC));
+		$hint = ($t = ObjectMixin::getSuggestion($items, $name)) ? ", did you mean $$t?" : '.';
+		throw new MemberAccessException("Attempt to read undeclared property {$rc->getName()}::$$name$hint");
+	}
+
+
+	/**
+	 * Access to undeclared property.
+	 * @throws MemberAccessException
+	 */
+	public function __set($name, $value)
+	{
+		$rc = new \ReflectionClass($this);
+		$items = array_diff($rc->getProperties(\ReflectionProperty::IS_PUBLIC), $rc->getProperties(\ReflectionProperty::IS_STATIC));
+		$hint = ($t = ObjectMixin::getSuggestion($items, $name)) ? ", did you mean $$t?" : '.';
+		throw new MemberAccessException("Attempt to write to undeclared property {$rc->getName()}::$$name$hint");
+	}
+
+
+	/**
+	 * @return bool
+	 */
+	public function __isset($name)
```

Comment:
You are right that `return FALSE` is useless a method can be removed. I'll remove it.